### PR TITLE
[RFF-2243]: custom component for initiator in OOTB payments journey

### PR DESCRIPTION
**Pull request by santam85**

### CODE REVIEW

File: apps/golden-sample-app/src/app/app.module.ts [hunk 1]
```diff
@@ -5,12 +5,15 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { ACCESS_CONTROL_BASE_PATH } from '@backbase/data-ang/accesscontrol';
 import { ARRANGEMENT_MANAGER_BASE_PATH } from '@backbase/data-ang/arrangements';
 import { TRANSACTIONS_BASE_PATH } from '@backbase/data-ang/transactions';
+import { CONTACT_MANAGER_BASE_PATH } from '@backbase/contact-manager-http-ang';
+import { PAYMENT_ORDER_BASE_PATH } from '@backbase/payment-order-http-ang';
```
[the remark this answers is not in the document]
Comment (yenobi) [reply apps/golden-sample-app/src/app/app.module.ts]: updated path

File: apps/golden-sample-app/src/app/custom-payment/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Add a custom component to Initiate Payments Journey
+
+## Scope
+
+Scope of this example is to explain how to create a custom component and connect it with the out-of-the-box Initiate Payments Journey. The component in this example replaces the out-of-the-box `intiator` group but it can also be used to replace any other configuration group such as `counterparty`, `remittanceInfo` and `schedule` or add a new `additions` configuration group.
+
+## Initate Payments Journey
+
+Initiate Payment journey allows customers create or edit a payment. The journey is fully customizable with a TypeScript configuration to support different payment types. Check the documentation on backbase.io to learn more about internal workings of the journey and out-of-the-box configurations for different payment types.
```
[the remark this answers is not in the document]
Comment (yenobi) [reply apps/golden-sample-app/src/app/custom-payment/README.md:9]: added actual links

**Review by yenobi — APPROVED** [review]

**Review by yenobi — APPROVED** [review]

File: apps/golden-sample-app/src/app/app.component.html [hunk 3]
```diff
@@ -144,6 +144,29 @@
...
+          <bb-icon-ui
+            class="bb-layout__vertical-nav-item-icon"
+            name="verified-user"
+          ></bb-icon-ui>
+
+          <span
+            class="bb-layout__horizontal-nav-item-description"
+            i18n="Make a Payment Link@@main.make-a-payment.link.text"
+          >
+            Make A Payment
```
Comment (kirillsud) [inline apps/golden-sample-app/src/app/app.component.html:166]: WDYT, if we call it 'Internal transfer'?

File: apps/golden-sample-app/src/app/app-routing.module.ts [hunk 4]
```diff
@@ -73,6 +73,14 @@ const routes: Routes = [
       ),
     canActivate: [AuthGuard, UserContextGuard],
   },
+  {
+    path: 'make-a-payment',
```
Comment (kirillsud) [inline apps/golden-sample-app/src/app/app-routing.module.ts:77]:
```suggestion
    path: 'transfer-internal',
```

**Review by kirillsud — COMMENTED** [review]
Great example! I like the explanation in README - it is very clear and has a link to the community.

But I couldn't finish the transfer. Error 401. Do you have an idea why?

```json
{
  "message": "Unable to audit the initiated event. Business operation will not be carried out.",
  "errors":[]
}
```

File: apps/golden-sample-app/src/app/custom-payment/components/initiator/initiator.component.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  private setupInitiatorFormGroup(fields: InitiatorFields[]) {
+    fields.forEach((field: InitiatorFields) => {
+      this.group.addControl(
+        field,
+        new FormControl(
+          '',
+          this.options.validators || [],
+          this.options.asyncValidators || []
```
[the remark this answers is not in the document]
Comment (yenobi) [reply apps/golden-sample-app/src/app/custom-payment/components/initiator/initiator.component.ts:101]: For the initiatior fields group (`id`, `name`, `accountNumber`, `currency`

File: apps/golden-sample-app/src/app/custom-payment/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- `counterparty`
+- `remittanceInfo`
+- `schedule`
+- `additions`(Optional)
+
+All out-of-the-box configurations define the above mandatory fields and in addition may have some fields specific to the payment type. These fields are combined together to create a payment config of type `InitiatePaymentConfig` provided to the journey using Injection Token `INITIATE_PAYMENT_CONFIG`. This is demonstrated in `initiate-payment-journey-bundle-module.ts`.
+
+## Custom Initiator Component
+
+The task is to replace the existing component used inside the `initiator` configuration group and replace it with a custom component. The example is valid not only for `initiator` group, but for any of the group of any of the component, current documentation is aimed to show the general principals steps-by-step.
```
[the remark this answers is not in the document]
Comment (yenobi) [reply apps/golden-sample-app/src/app/custom-payment/README.md]: fixed

File: apps/golden-sample-app/src/app/app.component.html (hunk 3 above)
Comment (yenobi) [reply apps/golden-sample-app/src/app/app.component.html]: updated with your proposal to be consistent

File: apps/golden-sample-app/src/app/app-routing.module.ts (hunk 4 above)
Comment (yenobi) [reply apps/golden-sample-app/src/app/app-routing.module.ts]: updated

**Review by yenobi — APPROVED** [review]

**Review by kirillsud — DISMISSED** [review]

**Review by yenobi — APPROVED** [review]

**Review by kirillsud — APPROVED** [review]

**Review by yenobi — APPROVED** [review]

**Pull request merged**